check-style.sh

```diff
@@ -10,8 +11,8 @@ function clang_format() {
 }
 
 ec=0
-mapfile -t files < <(find src include examples tests -iname "*.cc" -o -iname "*.h")
-for file in "${files[@]}"; do
+source_files=$(find src include examples tests -type f \( -name '*.h' -o -name '*.cc' \))
```

Comment:
This will break for file names containing spaces.

Reply:
Why would you create filenames with spaces? @balamurugana

Reply:
Also main reason for this change is that bash shipped by default on macOS  doesn't support for `mapfile`

Reply:
Output of `find` command will output file names on which non-project files will be listed and those file names might have spaces. If `mapfile` is not available, we would need to create a temporary file for `find` output to work reliably.

```bash
tmpfile="tmpfile.$RANDOM"
find src include examples tests -iname "*.cc" -o -iname "*.h" > "$tmpfile"
ec=0
while read -r file; do
    if ! clang_format "$file"; then
        ec=255
    fi
done < "$tmpfile"
rm -f "$tmpfile"
```

Reply:
done updated using the `filename` approach